ci-operator/step-registry/openshift/microshift/e2e/greenboot/openshift-microshift-e2e-greenboot-commands.sh

```diff
@@ -15,23 +15,23 @@ if [[ -z "${GOOGLE_COMPUTE_ZONE}" ]]; then
   exit 1
 fi
 
-mkdir -p "${HOME}"/.ssh
-
-# gcloud compute will use this key rather than create a new one
-cp "${CLUSTER_PROFILE_DIR}"/ssh-privatekey "${HOME}"/.ssh/google_compute_engine
-chmod 0600 "${HOME}"/.ssh/google_compute_engine
-cp "${CLUSTER_PROFILE_DIR}"/ssh-publickey "${HOME}"/.ssh/google_compute_engine.pub
-echo 'ServerAliveInterval 30' | tee -a "${HOME}"/.ssh/config
-echo 'ServerAliveCountMax 1200' | tee -a "${HOME}"/.ssh/config
-chmod 0600 "${HOME}"/.ssh/config
-
-cp "${CLUSTER_PROFILE_DIR}"/pull-secret "${HOME}"/pull-secret
-
 gcloud auth activate-service-account --quiet --key-file "${CLUSTER_PROFILE_DIR}"/gce.json
 gcloud --quiet config set project "${GOOGLE_PROJECT_ID}"
 gcloud --quiet config set compute/zone "${GOOGLE_COMPUTE_ZONE}"
 gcloud --quiet config set compute/region "${GOOGLE_COMPUTE_REGION}"
 
+IP_ADDRESS="$(gcloud compute instances describe ${INSTANCE_PREFIX} --format='get(networkInterfaces[0].accessConfigs[0].natIP)')"
+
+mkdir -p "${HOME}"/.ssh
+cat << EOF > "${HOME}"/.ssh/config
+Host ${INSTANCE_PREFIX}
+  User rhel8user
+  HostName ${IP_ADDRESS}
+  IdentityFile ${CLUSTER_PROFILE_DIR}/ssh-privatekey
+  StrictHostKeyChecking no
```

Comment:
Do we really want `StrictHostKeyChecking no`?

[This is how I dealt with that while rehearsing](https://github.com/openshift/release/pull/37629/files#diff-7ac53ea47493d45e52119a6e23842e6b5fca1c8f0a0dc1481a0dd79e6e57be3fR30-R41) `microshift/e2e/`:
```
cat <<EOF >"${HOME}/.ssh/config"
Host ${IP_ADDRESS}
    IdentityFile ~/.ssh/google_compute_engine
    ServerAliveInterval 30
    ServerAliveCountMax 1200
EOF
chmod 0600 "${HOME}/.ssh/config"

gcloud compute ssh \
  --project "${GOOGLE_PROJECT_ID}" --zone "${GOOGLE_COMPUTE_ZONE}" \
  "rhel8user@${INSTANCE_PREFIX}" \
  --command "cat /etc/ssh/ssh_host_*_key.pub" | cut -d' ' -f1,2 | sed "s,^,${IP_ADDRESS} ," >>~/.ssh/known_hosts
```

Reply:
The server we are connecting to will not change throughout the entire test run and we are using an external IP, plus we need this in every step along the way as they run in different containers. The idea behind this setting is to avoid the prompt about storing the server's public key.
Even with the option set to no it will still display a warning(as it stores the key anyway) but allow the connection to go through. I can also change this to `accept-new` so that it does not prompt and still keep the connection denial if the key changes.

Reply:
There are at most 4 connections in the container that connects the most times and half of them happen immediately after the previous connection.
For the sake of security I will set it to the secure non-prompt option, but I don't think this is something that could ever happen. Still, it does not hurt.